gpcontrib/arenadata_toolkit/arenadata_toolkit--1.0.sql

```diff
@@ -129,3 +129,96 @@ BEGIN
 END$$
 LANGUAGE plpgsql VOLATILE
 EXECUTE ON MASTER;
+
+CREATE FUNCTION arenadata_toolkit.adb_get_relfilenodes(tablespace_oid OID)
+RETURNS TABLE (
+	segindex int2,
+	dbid int2,
+	datoid oid,
+	tablespace_oid oid,
+	relfilepath text,
+	relfilenode oid,
+	reloid oid,
+	size bigint,
+	modified_dttm timestamp without time zone,
+	changed_dttm timestamp without time zone
+)
+EXECUTE ON ALL SEGMENTS
+AS '$libdir/arenadata_toolkit', 'adb_get_relfilenodes'
+LANGUAGE C STABLE STRICT;
+
+CREATE VIEW arenadata_toolkit.__db_segment_files
+AS
+	WITH segfiles AS (
```

Comment:
Using WITH in a VIEW results in the fact that even with a filter, the entire subquery will still be executed completely first.